messaging/layers/segmentation/segmenter.go

```diff
@@ -129,70 +115,58 @@ func segmentMessage(newMessage *wakutypes.NewMessage, segmentSize int) ([]*wakut
 		if err != nil {
 			return nil, err
 		}
-		segmentMessage, err := replicateMessageWithNewPayload(newMessage, marshaledSegmentWithMetadata)
-		if err != nil {
-			return nil, err
-		}
 
-		segmentMessages = append(segmentMessages, segmentMessage)
+		segmentMessages = append(segmentMessages, marshaledSegmentWithMetadata)
 		index++
 	}
 
 	return segmentMessages, nil
 }
 
-// handleSegmentationLayer is capable of reconstructing the message from both complete and partial sets of data segments.
-// It has capability to perform forward error correction.
-func (s *MessageSender) handleSegmentationLayer(message *types.Message) error {
-	logger := s.logger.Named("handleSegmentationLayer").With(zap.String("hash", cryptotypes.HexBytes(message.TransportLayer.Hash).String()))
-
-	segmentMessage := &types.SegmentMessage{
+func (s *Segmenter) TryReconstruct(payload []byte, sigPubKey *ecdsa.PublicKey) ([]byte, error) {
```

Comment:
Does it really need a `try` word? If so, then any function that can return an error should have a `try` prefix 😄 
```suggestion
func (s *Segmenter) Reconstruct(payload []byte, sigPubKey *ecdsa.PublicKey) ([]byte, error) {
```

Reply:
Well, in this case, `Try` adds extra semantics, emphasizing that failure is expected in most cases. `TryReconstruct` would be called many times before the original message is re-assembled from segments. I don't have a strong opinion, though, as you said, error codes might be enough. I removed the `Try`.